swift/apple/FirezoneNetworkExtension/NetworkSettings.swift

```diff
@@ -57,10 +58,17 @@ class NetworkSettings {
     self.hasUnappliedChanges = true
   }
 
+  func setMatchDomains(_ matchDomains: [String]) {
+    self.matchDomains = matchDomains
+    self.hasUnappliedChanges = true
+  }
+
   func apply(
-    on packetTunnelProvider: NEPacketTunnelProvider, logger: Logger,
+    on packetTunnelProvider: NEPacketTunnelProvider?,
```

Comment:
@roop By making this argument an optional, we can guard-return at the top of the function and not have to do it in each of the places where this function is being called.

This is a common pattern we used a lot in Ruby and we use a similar pattern in Rust.

If it's alright with you I'd like to open another PR cleaning these guards up a little so that @conectado and I and others can maintain the codebase a little more easily.

At present I feel that the `guard`s everywhere make things hard to read. I might even go so far as to say we might want to "let it crash" and use `!` when we expect variables to be initialized. We could really use these crash reports while we're in beta. Otherwise, these things could fail silently and we have to receive a user feedback report, then go dig through logs to find out what happened.

Reply:
Opened #3013 as a start to refactoring.